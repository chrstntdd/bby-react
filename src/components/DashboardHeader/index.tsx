import React from 'react';
import { Link } from 'react-router-dom';

const logoutIcon = 'images/logout.svg';

import './dashboard-header.scss';

const DashboardHeader = props => {
  let userName;
  props.userData
    ? (userName = `${props.userData.firstName} ${props.userData.lastName}`)
    : (userName = 'Employee');
  return (
    <header id="dashboard-header">
      <h1>Quantified Dashboard</h1>
      <h2>Hello, {userName}</h2>
      <Link to="/logout">
        <button>
          <img src={logoutIcon} alt="" />
          <p>Sign out</p>
        </button>
      </Link>
    </header>
  );
};

export default DashboardHeader;
