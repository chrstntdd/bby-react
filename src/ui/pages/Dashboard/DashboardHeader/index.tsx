import React from 'react';
import { Link } from 'react-router-dom';

const logoutIcon = 'images/logout.svg';

import './dashboard-header.scss';

const DashboardHeader = props => {
  let userName;
  props.userData
    ? (userName = `${props.userData.firstName} ${props.userData.lastName}`)
    : (userName = 'Employee');
  return (
    <header id="dashboard-header">
      <h1>Quantified Dashboard</h1>
      <h2>Hello, {userName}</h2>
      <Link onClick={props.logoutUser} to="/">
        <button>
          <img src={logoutIcon} alt="" />
          <p>Sign out</p>
        </button>
      </Link>
    </header>
  );
};

export default DashboardHeader;
