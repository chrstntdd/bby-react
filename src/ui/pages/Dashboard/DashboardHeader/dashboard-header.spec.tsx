import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import createRouterContext from 'react-router-test-context';

import DashboardHeader from './';

describe('<DashboardHeader/>', () => {
  const props = {
    userData: {
      firstName: 'testFirst',
      lastName: 'testLast'
    }
  };
  it('should render without crashing', () => {
    const wrapper = shallow(<DashboardHeader {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  it("should render a greeting with the user's first and last name", () => {
    const wrapper = shallow(<DashboardHeader {...props} />);
    const greeting = wrapper
      .find('h2')
      .props()
      .children.join('');
    expect(greeting).toBe(`Hello, ${props.userData.firstName} ${props.userData.lastName}`);
  });

  describe('the Logout button', () => {
    it('should render a button to direct back to the root route', () => {
      const wrapper = shallow(<DashboardHeader {...props} />);
      const logoutButton = wrapper.find('Link');
      expect(logoutButton.props().to).toBe('/');
      expect(logoutButton.props().children.type).toBe('button');
    });
  });
});
