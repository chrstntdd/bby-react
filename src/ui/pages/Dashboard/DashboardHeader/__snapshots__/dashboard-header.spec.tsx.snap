// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardHeader/> should render without crashing 1`] = `
<header
  id="dashboard-header"
>
  <h1>
    Quantified Dashboard
  </h1>
  <h2>
    Hello, 
    testFirst testLast
  </h2>
  <Link
    replace={false}
    to="/"
  >
    <button>
      <img
        alt=""
        src="images/logout.svg"
      />
      <p>
        Sign out
      </p>
    </button>
  </Link>
</header>
`;
